services/uptimerobot/uptimerobot-base.js

```diff
@@ -0,0 +1,97 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+const { InvalidParameter, InvalidResponse } = require('../errors')
+
+// https://uptimerobot.com/api
+// POST getMonitors
+const errorResponse = Joi.object({
+  stat: Joi.equal('fail').required(),
+  error: Joi.object({
+    message: Joi.string(),
+  }).default({}),
+}).required()
+
+const monitor = Joi.object({
+  status: Joi.equal(0, 1, 2, 8, 9).required(),
+})
+
+const monitorWithUptime = monitor.keys({
+  custom_uptime_ratio: Joi.string()
+    .regex(/^\d*\.\d{3}$/)
+    .required(),
+})
+
+const singleMonitorResponse = Joi.alternatives(
+  errorResponse,
+  Joi.object({
+    stat: Joi.equal('ok').required(),
+    monitors: Joi.array()
+      .length(1)
+      .items(monitor)
+      .required(),
+  }).required()
+)
+
+const singleMonitorResponseWithUptime = Joi.alternatives(
+  errorResponse,
+  Joi.object({
+    stat: Joi.equal('ok').required(),
+    monitors: Joi.array()
+      .length(1)
+      .items(monitorWithUptime)
+      .required(),
+  }).required()
+)
+
+module.exports = class UptimeRobotBase extends BaseJsonService {
+  static get category() {
+    return 'other'
```

Comment:
Can we just return 'monitoring' here and remove the `category()` functions from the child classes?

Reply:
Ha! Yes. It's funny, I went looking for the category and couldn't find it. I was looking in the wrong place :)